src/section/section_update_barrier.rs

```diff
@@ -55,12 +55,12 @@ impl SectionUpdateBarrier {
         self.try_get_details(our_prefix)
     }
 
-    fn try_get_details(&mut self, our_prefix: &Prefix<XorName>) -> Option<SectionUpdateDetails> {
+    fn try_get_details(&self, our_prefix: &Prefix<XorName>) -> Option<SectionUpdateDetails> {
```

Comment:
What's the reason for this change? The idea was that once we collect all the events, we return them and reset the barrier into its default state. Now we are keeping the events in forever? Seems like it can cause issues when old events are mixed with new ones, no?

Reply:
Barrier will still be reset after updating key got completed.
The reason of this change is that:

1, in case we get the SectionInfo first, then complete the DKG, the taken events from barrier will be `lost` even the later on update of section key got failed. Which means the node lost the chance to recover.

Reply:
1. Ok, I see that you are resetting it in `update_our_section`, so that's fine. Maybe in the future we can try to refactor it so that the barrier also waits for the DKG result. With that, we might be able to also remove the pending events cache. But I'm OK with this for now.
2. Yeah, good point. It should have been called `try_take_details`.